app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragment.kt

```diff
@@ -119,16 +119,18 @@ class ContactDetailsFragment : Fragment() {
         }
 
         view.findViewById<Button>(R.id.run_button).setOnClickListener {
-            val route = DUMMY_ROUTE[0] //remove when merged with contact creation and use real route
             val uri =
-                Uri.parse("google.navigation:q=${route.dst.latitude},${route.dst.longitude}&mode=b")
+                Uri.parse("google.navigation:q=${currentContact!!.latitude},${currentContact!!.longitude}&mode=b")
             val intent = Intent(Intent.ACTION_VIEW, uri)
             intent.setPackage(RouteFragment.MAPS_PKG)
             requireContext().startActivity(intent)
         }
 
         view.findViewById<Button>(R.id.show_all_button).setOnClickListener {
-            routeViewModel.addRoute(DUMMY_ROUTE[0]) //remove when merged with contact creation and use real route
+            if (currentContact!!.latitude == null || currentContact!!.longitude == null) {
```

Comment:
For that new latitude and longitude arguments, I remark it now and it should have been implemented in #174,
as we will have that check very often in the whole app, it could be really useful to have an  `hasCoordinates(): Boolean` function in the `Contact` data class